core/src/main/java/org/elasticsearch/client/transport/TransportClient.java

```diff
@@ -168,9 +172,17 @@ public TransportClient build() {
                 final TransportProxyClient proxy = new TransportProxyClient(settings, transportService, nodesService,
                     actionModule.getActions().values().stream().map(x -> x.getAction()).collect(Collectors.toList()));
 
+                List<LifecycleComponent> pluginLifecycles = pluginComponents.stream()
```

Comment:
Can you change `pluginLifecycles` to `pluginLifecycleComponents` or something. Plugins don't go through lifecycles. I hope.